components/core/src/streaming_archive/writer/File.cpp

```diff
@@ -117,31 +128,14 @@ namespace streaming_archive { namespace writer {
                                                            unordered_set<logtype_dictionary_id_t>& segment_logtype_ids,
                                                            unordered_set<variable_dictionary_id_t>& segment_var_ids)
     {
-        size_t var_ix = 0;
+        // Add logtype to set
```

Comment:
for #2. I agree. there will be a memory overhead for holding an extra set to store the IDs but hopefully it will be very small given that there aren't that many logtypes in total. will run experiment on our benchmark to see how much memory it will consume

Reply:
I might actually have some ideas to replace the set with some better data structure. so I would prefer to keep the code as it is before doing other experiments to avoid throw-away work

Reply:
Alright